packages/go_router_builder/example/lib/shell_route_example.dart

```diff
@@ -0,0 +1,134 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+part 'shell_route_example.g.dart';
+
+void main() => runApp(App());
+
+class App extends StatelessWidget {
+  App({super.key});
+
+  @override
+  Widget build(BuildContext context) => MaterialApp.router(
+        routerConfig: _router,
+      );
+
+  final GoRouter _router = GoRouter(
+    routes: $appRoutes,
+    initialLocation: '/foo',
+  );
+}
+
+class HomeScreen extends StatelessWidget {
+  const HomeScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) => Scaffold(
+        appBar: AppBar(title: const Text('foo')),
+      );
+}
+
+@TypedShellRoute<MyShellRouteData>(
```

Comment:
I may have missed something, how would one specified navigator key for typedshellRoute and also the parent navigator key for typedGoRoute

Reply:
I think what @johnpryan had in mind was to add a static final or getter called `$navigatorKey` to the class extending `ShellRouteData` and `GoRouteData` and this will be grabbed by the `_decodeKey` and added as the navigatorKey / parentNavigatorKey (BTW I fixed a small bug here: [6eab717](https://github.com/flutter/packages/pull/3439/commits/6eab71755acacc3e13c73902cced93ca780d9aa1))

Reply:
>  static final or getter

It seems a bit not obvious. Why not just let it be a parameter into the 
`@TypedShellRoute<MyShellRouteData>` ?

Reply:
I agree it's not obvious. I assumed that you two had agreed on the approach beforehand and I just picked up the work where it was left off.
So if we go ahead with adding:
- `navigatorKey` to `TypedShellRoute`
- `parentNavigatorKey` to `TypedGoRoute`

I will first need to make a PR on go_router (since that's where these classes are defined) and then raise the minimum version of go_router in go_router_builder.

Could you confirm that it's ok?

Reply:
yes, that is ok. Sorry about the miscommunication, the original PR was still in its initial state, so there may be some issue that hasn't been discovered yet.

Reply:
No problem, I also should have double checked.
Here is the PR on go_router for the first step: https://github.com/flutter/packages/pull/3471

Reply:
Hey @chunhtai, actually passing the keys as parameters of `TypedShellRoute` / `TypedGoRoute` doesn't work because the keys need to be `const` and `GlobalKey<NavigatorState>()` is not...
So is it ok to go ahead with this PR as it is now and I'll make another one to revert https://github.com/flutter/packages/pull/3471?

Reply:
<img width="840" alt="image" src="https://user-images.githubusercontent.com/2680243/228992257-a54abb89-a375-4067-9f74-7f84aa5e48b2.png">

Reply:
And https://dart.dev/language/metadata